custom_components/localtuya/__init__.py

```diff
@@ -310,13 +297,53 @@ async def update_listener(hass, config_entry):
     await hass.config_entries.async_reload(config_entry.entry_id)
 
 
+async def async_remove_config_entry_device(
+    hass: HomeAssistant, config_entry: ConfigEntry, device_entry: DeviceEntry
+) -> bool:
+    """Remove a config entry from a device."""
+    dev_id = list(device_entry.identifiers)[0][1].split("_")[-1]
+
+    ent_reg = await er.async_get_registry(hass)
```

Comment:
May want to consider the changes in https://github.com/rospogrigio/localtuya/pull/842 being applied here to avoid
```
Detected integration that uses deprecated `async_get_registry` to access entity registry, use async_get instead
```

Reply:
Gosh, I didn't know about this. OK I'll use the new call.